test/runtests.jl

```diff
@@ -41,3 +41,33 @@ using Test, SerializationCaches, OrderedCollections
         @test isdir(set_up_cache_path(joinpath(tmp, "big/made/up/path")))
     end
 end
+
+# Test that when `in_memory_limit` is 0, deserializing an existing file does not
+# (redundantly) call `put`
+function Base.put!(cache::SerializationCache, key::AbstractString, item;
+                   directly_to_file::Bool=false)
+   throw(error("Should not hit this!"))
+end
+
+@testset "No in_memory_limit" begin
+    mktempdir() do tmp
+        in_memory_limit = 0
+        file_limit, file_gc_ratio = 10, 0.1
+
+        # Set up existing cache
+        range = 1:file_limit
+        for i in range
+            serialize(joinpath(tmp, "$i.jls"), i)
+        end
+
+        # Set up cache
+        cache = SerializationCache(tmp; in_memory_limit=0,
+                                   file_limit, file_gc_ratio)
```

Comment:
FYI the reason this is failing CI on Julia 1.3 is that implicit keyword argument assignment is a 1.5 feature.
```suggestion
                                   file_limit=file_limit, file_gc_ratio=file_gc_ratio)
```